### PR TITLE
run chaos-daemon in privileged by default

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: cwen0 <cwenyin0@gmail.com>

### What is changed and how does it work? 

Add a field to control `privileged` mode and run chaos-daemon in privileged by default. 

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml [hunk 1]
```diff
@@ -64,6 +64,9 @@ spec:
               value: {{ .Values.timezone | default "UTC" }}
             {{- end }}
           securityContext:
+            {{- if .Values.chaosDaemon.privileged }}
+              privileged: true
```
Comment (STRRL) [inline helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:68]: Please remove 2 spaces here. 
I think `privileged: true` should keep the same indent level with `capabilities`.

Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml]: Fixed

**Review by STRRL — APPROVED** [review]
LGTM

File: helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml [hunk 2]
```diff
@@ -64,6 +64,9 @@ spec:
               value: {{ .Values.timezone | default "UTC" }}
             {{- end }}
           securityContext:
+            {{- if .Values.chaosDaemon.privileged }}
+            privileged: true
+            {{- else }}
             capabilities:
```
Comment (WangXiangUSTC) [inline helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:70]: do we need to remove these capabilities when privileged is true

Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:70]: Sorry, we should keep some capabilities.

Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:70]: `SYS_PTRACE` must be kept and others can be removed.

Comment (cwen0) [reply helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:73]: Other capabilities are unnecessary when privileged is true, except `SYS_PTRACE`. I think we can delete them to make the code cleaner

**Review by STRRL — APPROVED** [review]

**Pull request merged**